Source/SIMPLib/CoreFilters/ExtractVertexGeometry.cpp

```diff
@@ -0,0 +1,375 @@
+  //    // Create the Choice Filter Parameter and add it to the list of parameters
+  //    parameters.push_back(SIMPL_NEW_CHOICE_FP("New Data Container Geometry", NewDataContainerGeometry, FilterParameter::Parameter, ExtractVertexGeometry, m_NewDCGeometryChoices, false));
+  //  }
+
+  {
+    m_ArrayHandlingChoices.push_back("Move Attribute Arrays");
+    m_ArrayHandlingChoices.push_back("Copy Attribute Arrays");
+    // Create the Choice Filter Parameter and add it to the list of parameters
+    parameters.push_back(SIMPL_NEW_CHOICE_FP("Array Handling", ArrayHandling, FilterParameter::Parameter, ExtractVertexGeometry, m_ArrayHandlingChoices, false));
+  }
+
+  {
+    DataContainerSelectionFilterParameter::RequirementType req;
+    req.dcGeometryTypes = IGeometry::Types(2);
+    req.dcGeometryTypes[0] = IGeometry::Type::Image;
+    req.dcGeometryTypes[0] = IGeometry::Type::RectGrid;
+    parameters.push_back(SIMPL_NEW_DC_SELECTION_FP("Data Container with Input Geometry", SelectedDataContainerName, FilterParameter::RequiredArray, ExtractVertexGeometry, req));
+  }
+
+  {
+    MultiDataArraySelectionFilterParameter::RequirementType req;
+    req.amTypes = AttributeMatrix::Types(1, AttributeMatrix::Type::Cell);
+    parameters.push_back(SIMPL_NEW_MDA_SELECTION_FP("Included Attribute Arrays", IncludedDataArrayPaths, FilterParameter::RequiredArray, ExtractVertexGeometry, req));
+  }
+
+  parameters.push_back(SIMPL_NEW_DC_CREATION_FP("Vertex Data Container Name", VertexDataContainerName, FilterParameter::CreatedArray, ExtractVertexGeometry));
+
+  setFilterParameters(parameters);
+}
+
+// -----------------------------------------------------------------------------
+//
+// -----------------------------------------------------------------------------
+void ExtractVertexGeometry::dataCheck()
+{
+  setErrorCondition(0);
+  setWarningCondition(0);
+
+  if(m_ArrayHandling < 0)
+  {
+    QString ss = QObject::tr("'Array Handling' has not been selected.");
+    setErrorCondition(-2002);
+    notifyErrorMessage(getHumanLabel(), ss, getErrorCondition());
+    return;
+  }
+
+  if(m_ArrayHandling > m_ArrayHandlingChoices.size() - 1)
+  {
+    QString ss = QObject::tr("'Array Handling' index is out of bounds.");
+    setErrorCondition(-2003);
+    notifyErrorMessage(getHumanLabel(), ss, getErrorCondition());
+    return;
+  }
+
+  if(m_SelectedDataContainerName.isEmpty())
+  {
+    QString ss = QObject::tr("'Data Container with Input Geometry' is empty.  Please input a value.");
+    setErrorCondition(-2004);
+    notifyErrorMessage(getHumanLabel(), ss, getErrorCondition());
+    return;
+  }
+
+  DataContainer::Pointer dc = getDataContainerArray()->getPrereqDataContainer<AbstractFilter>(this, m_SelectedDataContainerName);
+  if(getErrorCondition() < 0)
+  {
+    return;
+  }
+
+  if(getVertexDataContainerName().isEmpty())
+  {
+    QString ss = QObject::tr("'New Data Container Name' is empty.  Please input a value.");
```

Comment:
Message should be updated to use Vertex Data Container Name instead of New Data Container Name.